src/ctapipe/image/muon/fitting.py

```diff
@@ -75,16 +82,25 @@ def taubin_circle_fit(x, y, mask):
     x_masked = x[mask]
     y_masked = y[mask]
 
+    if weights is None:
+        weights_masked = np.ones(len(x_masked))
+    else:
+        weights_masked = weights[mask]
+
     R = x.max()  # x.max() just happens to be identical with R in many cases.
```

Comment:
Looking at this, using half the field of view as initial guess for the ring radius doesn't make too much sense to me, it works for LST, but for the MSTs and SSTs, this results in an unphysically large initial guess.

Should we just use 1 degree? Could also be an keyword argument like the initial guess now you added for x and y.

Reply:
Also, the `CameraGeometry` has a radios method, which should just be used here I think.
Maybe also as an argument `fov_radius`.

Reply:
right, in my suggestion above instead of `(x,y)` `(x_masked, y_masked)` should be used, this will be then initial ring dependent. Or, indeed, just 1 degree, which is physically about right. Also, the condition on the possible values of fitted radius below should be updated accordingly.